src/odemis/gui/cont/tabs/fastem_main_tab.py

```diff
@@ -0,0 +1,201 @@
+        # need to wait for the tab to be shown on the first time.
+        self._initialized_after_show = False
+
+        # View Controller
+        self.vp = panel.vp_main
+        assert isinstance(self.vp, FastEMMainViewport)
+        vpv = collections.OrderedDict(
+            [
+                (
+                    self.vp,
+                    {
+                        "name": "Acquisition",
+                        "stage": main_data.stage,  # add to show crosshair
+                        "cls": guimod.StreamView,
+                        "stream_classes": EMStream,
+                    },
+                ),
+            ]
+        )
+        self.view_controller = viewcont.ViewPortController(tab_data, panel, vpv)
+
+        # Single-beam SEM stream
+        hwemt_vanames = ("resolution", "scale", "horizontalFoV")
+        emt_vanames = ("dwellTime",)
+        hwdet_vanames = ("brightness", "contrast")
+        hwemtvas = set()
+        emtvas = set()
+        hwdetvas = set()
+        for vaname in getVAs(main_data.ebeam):
+            if vaname in hwemt_vanames:
+                hwemtvas.add(vaname)
+            if vaname in emt_vanames:
+                emtvas.add(vaname)
+        for vaname in getVAs(main_data.sed):
+            if vaname in hwdet_vanames:
+                hwdetvas.add(vaname)
+
+        sem_stream = FastEMSEMStream(
+            "Single Beam",
+            main_data.sed,
+            main_data.sed.data,
+            main_data.ebeam,
+            focuser=main_data.ebeam_focus,
+            hwemtvas=hwemtvas,
+            hwdetvas=hwdetvas,
+            emtvas=emtvas,
+        )
+        tab_data.streams.value.append(sem_stream)  # it should also be saved
+        tab_data.semStream = sem_stream
+
+        # Handle the parent panel's size event and set the size of children panel accordingly
+        panel.pnl_user_settings.Bind(wx.EVT_SIZE, self.on_pnl_user_settings_size)
+        panel.pnl_tabs.Bind(wx.EVT_SIZE, self.on_pnl_tabs_size)
+
+        user_settings_panel = main_xrc.xrcpnl_fastem_user_settings(panel.pnl_user_settings)
+        self.user_settings_panel = FastEMUserSettingsPanel(user_settings_panel, main_data)
+
+        overview_panel = main_xrc.xrcpnl_tab_fastem_overview(panel.pnl_tabs)
+        self.overview_tab = FastEMOverviewTab(
+            "fastem_overview",
+            panel.btn_tab_overview,
+            overview_panel,
+            main_frame,
+            main_data,
+            self.vp,
+            self.view_controller,
+            sem_stream,
+        )
+
+        acquisition_panel = main_xrc.xrcpnl_tab_fastem_acqui(panel.pnl_tabs)
+        self.acquisition_tab = FastEMAcquisitionTab(
+            "fastem_acq",
+            panel.btn_tab_acqui,
+            acquisition_panel,
+            main_frame,
+            main_data,
+            self.vp,
+            self.view_controller,
+            sem_stream,
+        )
+
+        self._tab = model.VAEnumerated(None, choices={None: ""})
```

Comment:
Please create a sub class of MicroscopyGUIData adapted to this tab, and move this VA there. I'd suggest something like `settings_tab` or `active_tab` .